grpc-native/src/main/java/org/ballerinalang/net/grpc/ServiceDefinition.java

```diff
@@ -184,7 +197,50 @@ public ServiceDefinition(String rootDescriptor, BMap<BString, Object> descriptor
         return Collections.unmodifiableMap(descriptorMap);
     }
 
+    private Type getStreamDataType(Descriptors.MethodDescriptor methodDescriptor, MethodType attachedFunction,
+                                   String typeOfStream) {
+
+        Type functionReturnType = attachedFunction.getType().getReturnParameterType();
+        UnionType unionReturnType = (UnionType) functionReturnType;
+        Type streamParameterType = unionReturnType.getMemberTypes().get(0);
+        if (streamParameterType instanceof ErrorType && unionReturnType.getMemberTypes().size() > 1) {
```

Comment:
Does streamParamType come as the second param in unionType?

Reply:
I thought it might happen if someone specifies a remote function as below:
```ballerina
remote function nestedMsgServerStreaming(string value) returns error|stream<NestedMsg, error?>
```
Isn't that the case?

Reply:
I read it wrong. you covered both cases.